### DISCUSSION TITLE
can we switch to a more reliable encryption service?

### DISCUSSION THREAD

**Discussion by braindevices** [gsantner/markor#2216] [Ideas]
the jpencconverter seems not actively maintained since 2022.
I also cannot find any security audit about it.

Maybe it is better to switch to another encryption lib/service

Why not just use the openkeychain? which offer the gpg api for other app. (The project is stable and actively maintained for bug fix, although it does not add more feature)

I know little about the android development. So I do not know if there is OS level api can be used directly instead of relying on stuff like 3rd part libraries. Even if there is, we probably still want to use an encryption method that allow other OS to easily decrypt the file. gpg has support on almost any platform so I think it is still the best choice.

**Comment by braindevices** [comment]
although password-store has move to [pgpainless](https://github.com/pgpainless/pgpainless)
But the k9 mail remains to use openkeychain. Because there is going to [need lots of extra effort to create UI that openkeychain already provided](https://github.com/thunderbird/thunderbird-android/issues/5662)

So I guess the effortless choice here is still openkeychain

But if we want less effort in total we may want to invest into pgpainless

**Comment by gsantner** [comment]
Feel free to implement and keep maintaing something else/in addition for Markor.

So far AES256 hasn't been broken to my knowledge. The implementation uses standard java/android cryptography functionalities. 

No change since X doesn't mean it's unmaintained, it also can mean it is finished and does the job.

> cannot find any security audit about it.

Feel free to pay an audit for that library.

> Why not just use the openkeychain? which offer the gpg api for other app.


Being in control of what happens when. not relying on another app with async workflow and wait time for every single file that users have.

**Reply by braindevices** [reply]
I think it is not maintained because there is no reply to the issues.

**Reply by gsantner** [reply]
Ah OK. So as said, feel free to add support for another way of encrypting files - though make sure not to break existing encryption configuration for users (so they can still decrypt/encrypt & read files) and don't get locked out of their stuff.

**Comment by clach04** [comment]
Cross referencing to https://github.com/gsantner/markor/discussions/2538

@braindevices is this (alternative encryption) a project you are thinking about working on? The jenc format seems **OK** for my use, but I'm interested in alternatives. That being said I'm expressly NOT interested in anything OpenPGP based. If you have the time, I can recommend a short reading list that can explain (way better than I can) a few possible reasons for avoiding it:

  * https://soatok.blog/2024/11/15/what-to-use-instead-of-pgp/
  * https://www.latacora.com/blog/2019/07/16/the-pgp-problem/
